docs/data/data-grid/editing-legacy/editing.md

```diff
@@ -0,0 +1,272 @@
+
+### Server-side validation
+
+Server-side validation works like [client-side validation](#client-side-validation).
+The only difference is that when `preProcessEditCellProps` is called, a promise must be returned.
+Once the value is validated in the server, the promise should be resolved with a new object containing the `error` attribute set to true or false.
+The grid will wait for the promise to be resolved before exiting the edit mode.
+
+<!-- TODO v6: make the following paragraph the default behavior -->
+
+> By default, `preProcessEditCellProps` is called on each value change and during commit.
+> With column types that automatically commit the value after a selection (e.g. `singleSelect`) this means that the callback will be called twice and with the wrong value in the second call.
+> To avoid this inconsistency, enable the `preventCommitWhileValidating` flag available in the experimental features.
+>
+> ```jsx
+> <DataGrid experimentalFeatures={{ preventCommitWhileValidating: true }} />
+> ```
+>
+> It's labeled as experimental because in v6 this flag will become the default behavior.
+
+This demo shows how you can validate a username asynchronously and prevent the user from committing the value while validating.
+It's using `DataGridPro` but the same approach can be used with `DataGrid`.
+
+{{"demo": "ValidateServerNameGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Server-side persistence
+
+If you want to send the updated data to your server, you can use the `onCellEditCommit` which is fired just before committing the new cell value to the grid.
+
+You can then decide if you want to send the whole row or only the modified fields.
+
+{{"demo": "CellEditServerSidePersistence.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom edit component
+
+To customize the edit component of a column, use the `renderEditCell` attribute available in the `GridColDef`.
+
+The demo lets you edit the ratings by double-clicking the cell.
+
+{{"demo": "RenderRatingEditCellGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Edit using external button [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+You can override the default [start editing](#start-editing) triggers using the [`event.defaultMuiPrevented`](/components/data-grid/events#disabling-the-default-behavior) on the synthetic React events.
+
+{{"demo": "StartEditButtonGrid.js", "bg": "inline", "disableAd": true}}
+
+### Events [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+The editing feature leverages the event capability of the grid and the apiRef.
+The following events can be imported and used to customize the edition:
+
+- `cellEditStart`: emitted when the cell turns to edit mode.
+- `cellEditStop`: emitted when the cell turns back to view mode.
+- `cellEditCommit`: emitted when the new value is committed.
+- `editCellPropsChange`: emitted when the props passed to the edit cell component are changed.
```

Comment:
This is an odd usage of the word "emitted," which generally connotes a sensation like a smell or a noise. 

I think "triggered" would be more accurate.

```suggestion
- `cellEditStart`: triggered when a cell enters edit mode.
- `cellEditStop`: triggered when a cell returns to view mode.
- `cellEditCommit`: triggered when a new value is committed.
- `editCellPropsChange`: triggered when the props passed to the edit cell component are changed.
```

Reply:
I think it is derived from the term "Event emitter" which is very common.
So `emit and event` feels right to me.
But here we are talking about the DOM interaction which fires the event, so I'm fine with "triggered when"

Reply:
Would work for me